src/Servers/Kestrel/Core/src/Internal/TlsConnectionFeature.cs

```diff
@@ -65,7 +66,21 @@ public SslProtocols Protocol
         set => _protocol = value;
     }
 
+    public SslStream SslStream
+    {
+        get => _sslStream;
+    }
+
     // We don't store the values for these because they could be changed by a renegotiation.
+
+    TlsCipherSuite? ITlsHandshakeFeature.NegotiatedCipherSuite { get => NegotiatedCipherSuite; }
+    // Split the writable property from the interface implementation because it doesn't make sense to allow setting null
+    internal TlsCipherSuite NegotiatedCipherSuite
+    {
+        get => _cipherSuite ?? _sslStream.NegotiatedCipherSuite;
+        set => _cipherSuite = value;
+    }
```

Comment:
Why is it settable? It doesn't seem to be available anywhere. It seems like you don't need this property at all, you could directly return _sslStream.NegotiatedCipherSuite.

Actually, only ClientCertificate is directly settable from the interface. None of these others need setters.

Reply:
Just trying to follow local conventions.  Always happy to delete setters.  Do you want me to clean up all of them?